clients/drcachesim/analysis_tool.h

```diff
@@ -156,6 +156,14 @@ template <typename RecordType> class analysis_tool_tmpl_t {
      */
     virtual bool
     print_results() = 0;
+    /**
+     * Notifies the analysis tool that the given trace \p quantum_id has ended.
+     */
+    virtual bool
+    notify_quantum_end(uint64_t quantum_id)
```

Comment:
I was debating whether some other name would be clearer: "heartbeat" or "time_series_snapshot".   Maybe query more people to see whether "quantum_end" or "time_series_snapshot" is preferred?

Reply:
Changed it to generate_quantum_result/generate_shard_quantum_result for now.